DFE-Digital/prepare-academy-conversions
# Cypress refactor

**Pull Request Description**

### Overview of Changes:

1. **Linting Enhancements:**
   - Resolved ESLint issues in the Cypress codebase, specifically addressing the "unsafe to chain" error.
   - Implemented explicit separation of Cypress commands and introduced `cy.wrap()` for improved clarity.

2. **Test Refactoring and Bug Resolution:**
   - Conducted comprehensive refactoring of existing Cypress tests, aiming for a 100% pass rate (Apart from a known issues, including Bug 151197 related to Link Field Validation)

3. **Enhanced Cypress Launch Process:**
   - Optimised the launch process for Cypress tests, simplifying execution for users.
   - Introduced a streamlined credential management approach using `Cypress.Env.json`.
   - Now, launching Cypress is as straightforward as executing `npx cypress open`.

4. **Code Cleanliness and Efficiency:**
   - Performed various refactors to enhance code cleanliness and efficiency.
   - This ensures a more maintainable and performant codebase.

### How to Verify Changes:

1. **Run Cypress Tests:**
   - Pull the latest changes from the repository.
   - Execute Cypress tests using the command `npx cypress open`.
   - Verify that all tests run successfully, confirming the enhanced stability.
